src/IO/ADIOS/CommonADIOS1IOHandler.cpp

```diff
@@ -311,8 +311,8 @@ CommonADIOS1IOHandlerImpl::flush_attribute(int64_t group, std::string const& nam
         }
         case DT::BOOL:
         {
-            auto ptr = reinterpret_cast< unsigned char* >(values.get());
-            *ptr = static_cast< unsigned char >(att.get< unsigned char >());
+            auto ptr = reinterpret_cast< bool* >(values.get());
+            *ptr = static_cast< bool >(att.get< bool >());
```

Comment:
`values` is a `void*` that is given to ADIOS, allocated [here](https://github.com/openPMD/openPMD-api/blob/dev/src/IO/ADIOS/CommonADIOS1IOHandler.cpp#L99).
to support bool writing, this will have to be allocated as `uint8_t`.
The `Attribute` is passed as `bool`, so this needs to be changed manually.

```cpp
void
CommonADIOS1IOHandlerImpl::flush_attribute(int64_t group, std::string const& name, Attribute const& att)
{
    Auto dtype = att.dtype;
    if( dtype == Datatype::BOOL )
        dtype = Datatype::UCHAR;

    switch( att.dtype )
    /* ... */

    std::unique_ptr< void, std::function< void(void*) > > values = auxiliary::allocatePtr(att.dtype, nelems);
    switch( att.dtype )
    {
        case DT::BOOL:
        {
            auto ptr = reinterpret_cast< unsigned char* >(values.get());
            *ptr = static_cast< unsigned char >(att.get< bool >());
            break;
        }
    }
    /* ... */
}
```

Reply:
@C0nsultant can you push the change you are suggesting. I am open to any changes here, just needed a demonstrator to show what I saw was causing the crash :)